src/main/java/org/spoken_tutorial/health/elasticsearch/controllers/HomeController.java

```diff
@@ -0,0 +1,294 @@
+            if (doesFileExist(documentPath)) {
+                queuemnt.setDocumentPath(documentPath);
+                queuemnt.setMessage("success");
+                Optional category1 = Optional.ofNullable(category);
+                if (category1.isPresent()) {
+                    queuemnt.setCategory((String) category1.get());
+                }
+
+                Optional topic1 = Optional.ofNullable(topic);
+                if (topic1.isPresent()) {
+                    queuemnt.setTopic((String) topic1.get());
+                }
+
+                Optional outlinePath1 = Optional.ofNullable(outlinePath);
+                if (outlinePath1.isPresent()) {
+                    if (doesFileExist((String) outlinePath1.get())) {
+                        queuemnt.setOutlinePth((String) outlinePath1.get());
+                    }
+
+                }
+
+                queRepo.save(queuemnt);
+                resultMap.put(queuemnt.getQueueId(), "success");
+            }
+
+            else {
+                resultMap.put(queuemnt.getQueueId(), "failed");
+            }
+        } else {
+            resultMap.put(null, "failed");
+        }
+
+        return resultMap;
+
+    }
+
+    @PutMapping("/updateDocumentRank/{documentId}/{documentType}/{rank}")
+    public Map<Long, String> updateDocumentRank(@PathVariable String documentId, @PathVariable String documentType,
+            @PathVariable int rank) {
+
+        QueueManagement queuemnt = queRepo.findByDocumentTypeAndDocumentId(documentType, documentId);
+        Map<Long, String> resultMap = new HashMap<>();
+
+        if (queuemnt != null) {
+            queuemnt.setRank(rank);
+            queuemnt.setMessage("success");
+            resultMap.put(queuemnt.getQueueId(), "success");
+
+        } else {
+            resultMap.put(null, "failed");
+        }
+
+        return resultMap;
+
+    }
+
+    @DeleteMapping("/deleteDocument/{documentType}/{documentTypeId}")
+    public Map<Long, String> deleteDocument(@PathVariable String documentId, @PathVariable String documentType) {
+
+        QueueManagement queuemnt = queRepo.findByDocumentTypeAndDocumentId(documentType, documentId);
+        Map<Long, String> resultMap = new HashMap<>();
+
+        if (queuemnt != null) {
+
+            queRepo.delete(queuemnt);
+            resultMap.put(queuemnt.getQueueId(), "success");
+        } else {
+            resultMap.put(null, "failed");
+        }
+
+        return resultMap;
+
+    }
+
+    /************
+     * Testing
+     ************************************************************/
+
+    String path = "D:\\Users\\aloks\\Documents\\TimeScript.pdf";
+
+    @PostMapping("/saveTest")
+    Map<Long, String> saveTest() {
+        Map<Long, String> resultMap = addDocument("1", "tutorial", path, "url", 1, "view_url", "English", "category",
+                "topic", null);
+        return resultMap;
+    }
+
+//    @PostMapping("/addDocumentTest/{content}/{documentType}/{documentTypeId}")
```

Comment:
remove commented code

if this is for testing, keep the code and don´t comment it.